pkg/xds/server/components.go

```diff
@@ -359,11 +370,22 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler, ingressRec
 					FaultInjections:    faultInjection,
 					CLACache:           claCache,
 				}
-				return reconciler.Reconcile(envoyCtx, &proxy)
+				err = reconciler.Reconcile(envoyCtx, &proxy)
+				if err == nil {
+					prevError = nil
+				}
+				return err
```

Comment:
instead of introducing `prevErr`. Can we do this
```
err = reconciler.Reconcile(envoyCtx, &proxy)
if err != nil {
  return err;
}
prevHash = snapshotHash
return nil
```